### DISCUSSION TITLE
RESTful api to mars-sim

### DISCUSSION THREAD

**Discussion by bevans2000** [mars-sim/mars-sim#473] [Ideas]
One direction to consider for the future is converting the core engine into a remote service deployed on the net that can be shared by multiple parallel clients.
To do this the obvious way is to create a RESTful API that can delivery the properties of Simulation units/Entities to a remote client. We have something basic by allowing the Console interface to be used on a remote SSH connection. But this is for human users and not client software, e.g. mobile or separate web site.
We need an API that can deliver JSON representations of the entities. This is not the same as the internal representation that maybe used to persistent objects.
One way we could do this is to look at expanding the currently Console Command framework. Mot commands use the ```StructuredResponse```  to format output in a structured manner to the user. However we could provide a different implementation that present the data as JSON instead of text. The Commands would then provide the binding between the remote clients and the simulation internals.

**Comment by bevans2000** [comment]
@mokun I forget to tag you on this idea.

**Comment by mokun** [comment]
> a remote service deployed on the net that can be shared by multiple parallel clients.

So can you also list out any particular near-term use cases or applications for connecting to multiple clients ?

**Reply by mokun** [reply]
@Urwumpe, any thoughts on that as well ?
